src/operator/tensor/dot.cc

```diff
@@ -51,13 +150,19 @@ NNVM_REGISTER_OP(dot)
     dot(x,y)[0,0,1,1] = 0
     sum(x[0,0,:]*y[:,1,1]) = 0
 
-The storage type of ``dot`` output depends on storage types of inputs and transpose options:
+The storage type of ``dot`` output depends on storage types of inputs, transpose options and given
+hint for output storage type:
 
+Implemented sprase operations include:
 - dot(csr, default) = default
-- dot(csr.T, default) = row_sparse
+- dot(csr, default, transpose_a=True) = row_sparse
 - dot(csr, row_sparse) = default
-- dot(default, csr) = csr
-- otherwise, ``dot`` generates output with default storage
+- dot(default, csr) = csr on CPU only
+- dot(default, csr) = dense on GPU only
```

Comment:
Is it better to write "dot(default, csr, forward_stype_hint='default')"?

Reply:
Also dot(default, csr, transpose_b=True, forward_stype_hint='default') ?

Reply:
Also just choose one of "default" or "dense".. Mixing them in the same line will be confusing

Reply:
Done